### ISSUE TITLE
The script never opens...

### ISSUE THREAD

**Issue by MacBucky** [talkingmoose/Outlook-Exchange-Setup-5#8]
Hi Folks,

I used the original version 5.0 of the Outlook exchange setup and Outlook 2016 and it worked great.  Now that I am attempting to move our infrastructure to El Capitan, I loaded the most recent version of this.  I configured the settings appropriately like I did with version 5.  We moved everything into the appropriate places.  When I log in as my AD user account, I open Outlook 2016 version 15.19.1 and nothing is happening.  I can open the applescript manually and it works, but it never launches on first run.  Is there a way to look at log files to see if the launch file is attempting to launch?  I looked in my user folder log locations and there is nothing there for talking moose.

It feels like I am simply missing something.  Am I Over looking something...thoughts?

mick

**Comment by MacBucky** [comment]
I should have left my email for replies if needed...
mconners@madisoncollege.edu

Thank you.

**Comment by talkingmoose** [comment]
Hi Mick!

I added logging in version 5.2.x. If things in are in place you should at least see something in the user's home folder:

~/Library/Logs/OutlookExchangeSetup5.log

The log file gets created when the launch agent in the top level /Library folder calls the shell script it references. If you're running 5.2.1 of the script and you don't see any log file at all then one or both of these files is either not in place or not executing.

/Library/LaunchAgents/net.talkingmoose.OutlookExchangeSetupLaunchAgent.plist
/Library/Talking Moose Industries/Scripts/OutlookExchangeSetupLaunchAgent.sh

Here's what you should find with the logging:

https://github.com/talkingmoose/Outlook-Exchange-Setup-5/wiki/Logging

**Comment by MacBucky** [comment]
It is now working, thank you for checking back so quickly.  I am running into an issue however that Outlook is automatically opening for ALL users.  I only want it to initiate the setup process when Outlook is opened the first time.  One shared Macs, Students only use the web interface for Office 365.  Staff and faculty use the application Outlook.  Any way of making that happen?

Mick

**Comment by TeakWoodKite** [comment]
Hello TalkingMoose I am working with my JAMF partner to implement this and had it working perfectly except for;
1) when the AD user logs in and starts the Outlook2016 client the first time is configures perfectly.
2) the user closes the Outlook 2016 
3) the user Re-opens Outlook 2016 again (for every time thereafter) it creates a duplicate account. I see from the script it is supposed to check if an outlook folder exists before running the create account routine. 
We see the folder does exist, yet the script still runs again and again. It does not appear to delete the first run script. 
Any suggestions? My JAMF rep and I have gone through this with a fine tooth comb. Many thanks.
I am starting from scratch again.

**Comment by MacBucky** [comment]
Hi William,

Sorry to hear the news.  I have not a single clue why this would be occurring.  Let me ask, do you have an account with Apple for support?  We have an enterprise Select account.  Perhaps I could open a ticket and send them the scripts and see if they can provide us some feedback?

Mick

**Comment by talkingmoose** [comment]
Hi TeakWoodKite!

If you're getting multiple accounts then the launch agent in the user's home folder at ~/Library/LaunchAgents/net.talkingmoose.OutlookExchangeSetup.plist is not getting deleted by the AppleScript. The AppleScript may be failing to complete.

Make sure you're running the latest version of the script (5.2.1). It includes logging and you should see a log file appear in ~/Library/Logs/OutlookExchangeSetup5.log. See if the log file tells you what may be failing.

**Comment by TeakWoodKite** [comment]
Thank you sir. I will check that condition.

**Comment by talkingmoose** [comment]
@MacBucky, I've seen the same behavior too where Outlook seems to auto launch at login. The launch agent that launches Outlook is set not to run at load, which tells me this is either an Apple bug with launchd or the other trigger (a change to the OutlookProfile.plist file) is running. It stopped happening for me during my testing.

If you could reproduce this a couple of times and then send me the **~/Library/Logs/OutlookExchangeSetup5.log** file generated in the user's home folder, I'm glad to take a look and see if I need to adjust something.

**Comment by talkingmoose** [comment]
I've added additional logging to indicate whether the files and folders the script touches exist or don't exist in the user's home folder. The log file is in ~/Library/Logs/OutlookExchangeSetup5.log.

Specifically, once an Exchange account has been created, the log file should report at each login:

"/Users/mmoose/Library/Group Containers/UBF8T346G9.Office folder already exists. Doing nothing."
"/Users/mmoose/Group Containers/UBF8T346G9.Office/OutlookProfile.plist already exists. Doing nothing."
"/Users/mmoose/Library/LaunchAgents already exists. Doing nothing."
"/Users/mmoose/Library/LaunchAgents/net.talkingmoose.OutlookExchangeSetup5.plist does not exist. Doing nothing."

If any of the above lines returns "Something may be wrong" then the specific folder or file either exists when it shouldn't or doesn't exist when it should.

**Comment by MacBucky** [comment]
Here is the log file using the latest scripting that you uploaded recently.

Mick

**Comment by MacBucky** [comment]
Now, it is appearing to work, although it is continuing to create the untitled accounts.

Mick

**Comment by talkingmoose** [comment]
@MacBucky, doesn't look like replying by email allows the log file to come through. Could you send it to me privately? I'm curious to see what the new logging shows.

**Comment by MacBucky** [comment]
Sure thing Bill.

What address would you want me to send it to?

**Comment by talkingmoose** [comment]
@MacBucky, use the email in my GitHub profile. That'll avoid you having to post potentially sensitive information in public.

**Comment by MacBucky** [comment]
I think I did actually do that originally.  Is there any way to delete the log file that was sent to that other address?  In the log file, my cell phone number was accidentally added without me knowing that.

Mick

**Comment by talkingmoose** [comment]
@MacBucky, looks like your attachment got stripped, so no log file to delete. If you could send me a new log file using the updated scripts (from yesterday), that should give me the information I'm needing.

**Comment by MacBucky** [comment]
That’s what I am trying to send Bill.

I compress it and try again.

**Comment by talkingmoose** [comment]
After some testing, I've found a couple of unusual scenarios where two launch agents appear in the main /Library/LaunchAgents folder instead of one in the /Library/LaunchAgents folder and the other in the user's ~/Library/LaunchAgents folder. (Enable and log in to OS X's root account and install the package or install the package via command line after using sudo -s.)

Therefore, I modified the postinstall script to not run the user's launch agent as root if installing by either of those methods or something similar.

This should take care of some issues that may be causing multiple untitled accounts and/or launching Outlook automatically.

You'll find v5.4.0 here:
https://github.com/talkingmoose/Outlook-Exchange-Setup-5

And the change log here:
https://github.com/talkingmoose/Outlook-Exchange-Setup-5/wiki/Change-log
